analysis/analyzer.py

```diff
@@ -0,0 +1,133 @@
+import pandas as pd
+from numpy import sum
+import sys
+
+
+class Options:
+    def __init__(self):
+        self.type = ''
+        self.ip = ''
+        self.state = ''
+        self.port = ''
+        self.ippref = False
+        self.protocol = ''
+        self.ip_version = 0  # 0 == any
+        self.des = ''
+
+    def reset(self):
+        self.type = ''
+        self.ip = ''
+        self.state = ''
+        self.port = ''
+        self.ippref = False
+        self.protocol = ''
+        self.ip_version = 0
+        self.des = ''
+
+
+# TODO: implement buffering for large files?
+def load(filename):
+    df = pd.read_csv(filename, names=['Type', 'IPAddress', 'Port/info', 'State', 'description'],
+                     header=None, error_bad_lines=False)
+    op = (df.State.values == 'open').sum()
+    warnings = (df['description'].str.startswith('Warning')).sum()
+    suspicious = (df.State.values == 'suspicious').sum()
+    n = len(pd.unique(df['IPAddress']))
+    print(len(df), "records,", n, "distinct addresses,", op, "open ports", suspicious, "suspicious entries,", warnings,
```

Comment:
Consider either returning these stats along with the dataframe or moving this analysis out to a separate function.

Reply:
It looks like this logic is repeated in `show()`